From d889411703b8a79edd2c78b478f6c4ea9c4cb5ae Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 29 Jul 2015 08:36:49 +0300
Subject: [PATCH 0285/1240] boot: create seperate bootcmd and bootargs for non
 emulation

The baudrate and kernel_addr should be different.
Also remove the use of ramfs for real SOC

Change-Id: I3ae40bf2dea1abf0ec56c842396a754fe36b080b
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/22292
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 include/configs/armada8k.h | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/include/configs/armada8k.h b/include/configs/armada8k.h
index 193fa89..3593edf 100644
--- a/include/configs/armada8k.h
+++ b/include/configs/armada8k.h
@@ -75,6 +75,11 @@ extern char __end_of_spl[];
  * Add here all config values that differ
  * from the generic value in mvebu-common.h
  */
+#undef CONFIG_EXTRA_ENV_SETTINGS
+#undef CONFIG_BOOTARGS
+#undef CONFIG_BOOTCOMMAND
+
+/* Emulation specific setting */
 #ifdef CONFIG_PALLADIUM
 
 #undef CONFIG_BOOTDELAY
@@ -83,20 +88,27 @@ extern char __end_of_spl[];
 #undef CONFIG_BAUDRATE
 #define CONFIG_BAUDRATE                 24000
 
-#undef CONFIG_BOOTARGS
 #define CONFIG_BOOTARGS		"console=ttyS0,24000 earlycon=uart8250,mmio32,0xf0512000 " \
 				"mem=128M init=/bin/sh root=/dev/ram0 rw"
 
-#undef  CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	"kernel_addr=0x180000\0"	\
 					"fdt_addr=0x1000000\0"		\
 					"ramfs_addr=0x3000000\0"	\
 					"fdt_high=0xa0000000\0"		\
 					"initrd_high=0xffffffffffffffff\0"
 
-#undef  CONFIG_BOOTCOMMAND
 #define CONFIG_BOOTCOMMAND		"booti $kernel_addr $ramfs_addr $fdt_addr"
 
+#else /* Real SOC settings */
+#define CONFIG_BOOTARGS		"console=ttyS0,115200 earlycon=uart8250,mmio32,0xf0512000 root=/dev/nfs rw"
+
+#define CONFIG_EXTRA_ENV_SETTINGS	"kernel_addr=0x2000000\0"	\
+					"fdt_addr=0x1000000\0"		\
+					"fdt_high=0xa0000000\0"		\
+					"initrd_high=0xffffffffffffffff\0"
+
+#define CONFIG_BOOTCOMMAND		"booti $kernel_addr - $fdt_addr"
+
 #endif /*CONFIG_PALLADIUM*/
 
 
-- 
1.9.1

